fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Scope.java

```diff
@@ -53,7 +54,12 @@ private Optional<ResolvedField> resolveField(SlotRef expression, int fieldIndexO
         if (matchFields.size() > 1) {
             throw new SemanticException("Column '%s' is ambiguous", expression.getColumnName());
         } else if (matchFields.size() == 1) {
-            return Optional.of(asResolvedField(matchFields.get(0), fieldIndexOffset));
+            if (matchFields.get(0).getType().getPrimitiveType().equals(PrimitiveType.UNKNOWN_TYPE)) {
+                throw new SemanticException("Column " + matchFields.get(0).getName()
```

Comment:
Would better add the `external table` prefix to the error msg

Reply:
> Would better add the `external table` prefix to the error msg

Added